src/medimail/medimail-client.service.ts

```diff
@@ -0,0 +1,248 @@
+  status: 'SUCCESS' | 'ERROR';
+  request: {
+    id: string;
+    xml: string;
+  };
+  response: {
+    formatted: MSSSoapResult;
+    xml: string;
+    error: Error | null;
+  };
+};
+
+type SendMessageOptions = {
+  title: string;
+  message: string;
+  signatories: string[];
+  recipients?: string[];
+  attachments?: SendAttachments;
+};
+
+type SendResult = {
+  status: string,
+  author: string,
+  signatories: string,
+  ignore: null,
+  title: string,
+  refs: {
+    mss: string[] | string
+  },
+}
+
+type Signatory = {
+  firstname: string;
+  lastname: string;
+  email: string;
+}
+
+type Recipient = {
+  firstname: string;
+  lastname: string;
+  email: string;
+}
+
+type OpenResult = {
+  status: string,
+  call: string,
+  author: {
+    firstname: string,
+    lastname: string,
+    email: string,
+  },
+  begindate: string,
+  enddate: string | null,
+  signatories: Signatory[],
+  recipients: Recipient[],
+  title: string,
+  content: string,
+  nbattachments: string,
+  attachments: Attachment[],
+  'message-id': string,
+  inReplyTo: string | null,
+  references: string | null,
+  replyTo: string,
+}
+
+export class MedimailClient {
+  private readonly _wsdlUrl: string = path.resolve(
+    __dirname,
+    '../wsdl/medimail.wsdl'
+  );
+
+  private _soapClient: Client;
+  private _httpClient: HttpClient;
+  private _wsdl: WSDL;
+  private _acount: string;
+
+  constructor() { }
+
+  public async init(pfx: Buffer, passphrase: string = '', acount = ''): Promise<void> {
+    this._httpClient = new HttpClient();
+    this._acount = acount;
+    this._wsdl = new WSDL(this._wsdlUrl, this._wsdlUrl, {});
+    this._soapClient = await createClientAsync(this._wsdlUrl, {
+      forceSoap12Headers: true,
+      httpClient: this._httpClient,
+    });
+    this._soapClient.setSecurity(
+      new ClientSSLSecurityPFX(pfx, {
+        passphrase,
+      })
+    );
+  }
+
+  /**
+   * Test the api with a simple hello.
+   *
+   * @param name Name of the person to greet
+   * @returns complete once we get the first successful response
+   */
+  public async hello(name: string) {
+    const payload = { name };
+    return this._call(MedimailActions.HELLO, payload);
+  }
+
+  /**
+   * Send a message to the Medimail API.
+   *
+   * @param message The message to send
+   * @returns complete once we get the first successful response
+   */
+  public async send(options: SendMessageOptions): Promise<SendResult> {
+    const { title, message, signatories, recipients,attachments } = options;
+    const joinedSignatories = signatories.join(';');
+    const joinedRecipients = recipients?.join(';');
+    const payload = {
+      message,
+      title,
+      acount: this._acount,
+      signatories: joinedSignatories,
+      recipients: joinedRecipients,
+      attachments,
+    };
+    const reply = await this._call(MedimailActions.SEND, payload);
+    return this._wsdl.xmlToObject(reply.response.formatted.sendResult.$value).webisend;
```

Comment:
We are not guaranteed that there is no extra data we might need. It is safer to return an object with a one of its key being the actual json. It will be painful to move later on.